frontend/src/pages/HomePage.jsx

```diff
@@ -1,13 +1,151 @@
 import { NavLink, useLoaderData } from "react-router-dom";
+import { useState } from "react";
 
 function HomePage() {
   const recipes = useLoaderData();
+
+  const difficultyEmoji = (difficulty) => {
+    switch (difficulty) {
+      case 1:
+        return "🧑🏻‍🍳";
+      case 2:
+        return "🧑🏻‍🍳🧑🏻‍🍳";
+      case 3:
+        return "🧑🏻‍🍳🧑🏻‍🍳🧑🏻‍🍳";
+      default:
+        return "❓";
+    }
+  };
+
+  const checkTime = (time, cooking) => {
+    const timePreparation = parseInt(time, 10);
+    switch (cooking) {
+      case "10-20":
+        return timePreparation >= 10 && timePreparation <= 20;
+      case "20-30":
+        return timePreparation > 20 && timePreparation <= 30;
+      case "35+":
+        return timePreparation > 35;
+      default:
+        return false;
+    }
+  };
+
+  const [filteredRecipes, setFilteredRecipes] = useState(recipes);
+  const [searchText, setSearchText] = useState("");
+  const [selectedType, setSelectedType] = useState("All");
+  const [selectedDifficulty, setSelectedDifficulty] = useState("");
+  const [selectedTime, setSelectedTime] = useState("");
+
+  const filterRecipes = (text, type, difficulty, time) => {
+    const updatedRecipes = recipes.filter(
+      (r) =>
+        (type === "All" || r.type === type) &&
+        (difficulty === "" || r.difficulty.toString() === difficulty) &&
+        (time === "" || checkTime(r.preparation_time, time))
+    );
+    setFilteredRecipes(updatedRecipes);
+  };
+
+  const handleSearch = (e) => {
+    const text = e.target.value;
+    setSearchText(text);
+    const searchData = recipes.filter((d) =>
+      d.title.toLowerCase().includes(text.toLowerCase())
+    );
+    setFilteredRecipes(searchData);
+  };
+
+  const handleFilter = (e) => {
```

Comment:
Could you change the name of the function parameter to make it more understandable ?